File: Java-OOP-Basics/Problems Course Instance 2016_2017/src/Lesson03Inheritance/Exercise/pr03_mankind/Student.java
package Lesson03Inheritance.Exercise.pr03_mankind;

public class Student extends Human {

    private static final String INVALID_FACULTY_NUMBER = "Invalid faculty number!";

    private String facultyNumber;

    public Student(String firstName, String lastName, String facultyNumber) {
        super(firstName, lastName);
        this.setFacultyNumber(facultyNumber);
    }

    private void setFacultyNumber(String facultyNumber) {
        if (facultyNumber == null || facultyNumber.trim().length() < 5 || facultyNumber.trim().length() > 10) {
            throw new IllegalArgumentException(INVALID_FACULTY_NUMBER);
        }

        this.facultyNumber = facultyNumber;
    }

    @Override
    protected String getInfo() {
        return String.format("%s%nFaculty number: %s", super.getInfo(), this.facultyNumber);
    }
}

File: C#-OOP-Basics/Problems Course Instance 2018/Inheritance/Mankind/Student.cs
using System;
using System.Linq;
using System.Text;

public class Student : Human
{
    private string facultyNumber;

    public Student(string firstName, string lastName, string facNum)
        : base(firstName, lastName)
    {
        this.FacNum = facNum;
    }

    public string FacNum
    {
        get
        {
            return this.facultyNumber;
        }

        private set
        {
            if (!value.All(char.IsLetterOrDigit))
            {
                throw new ArgumentException("Invalid faculty number!");
            }

            if (value.Length < 5 || value.Length > 10)
            {
                throw new ArgumentException("Invalid faculty number!");
            }

            this.facultyNumber = value;
        }
    }

    public override string ToString()
    {
        StringBuilder sb = new StringBuilder();
        sb.Append(base.ToString());
        sb.AppendLine($"Faculty number: {this.facultyNumber}");

        return sb.ToString();
    }
}